tsdb/store.go

```diff
@@ -566,6 +566,7 @@ func (s *Store) DeleteDatabase(name string) error {
 		// no files locally, so nothing to do
 		return nil
 	}
+	sfile := s.sfiles[name]
```

Comment:
This is racy with another goroutine creating a new database. You should use `seriesFile(database string)`.

Reply:
L563 takes an `RLock` which is what `seriesFile()` does I believe.  This looks ok to me.

Reply:
Missed that `RLock`.